mbuild/packing.py

```diff
@@ -313,6 +320,156 @@ def fill_region(compound, n_compounds, region, overlap=0.2,
     return filled
 
 
+def fill_sphere(compound, sphere, n_compounds=None, density=None, overlap=0.2,
+                seed=12345, edge=0.2, compound_ratio=None,
+                fix_orientation=False, temp_file=None):
+    """Fill a sphere with a compound using packmol.
+
+    One argument of `n_compounds and density` must be specified.
+
+    If `n_compounds` is not None, the specified number of
+    n_compounds will be inserted into a sphere of the specified size.
+
+    If `density` is not None, the corresponding number of
+    compounds will be calculated internally.
+
+    Parameters
+    ----------
+    compound : mb.Compound or list of mb.Compound
+        Compound or list of compounds to be put in box.
+    sphere : list, units nm
+        Sphere coordinates in the form [x_center, y_center, z_center, radius]
+    n_compounds : int or list of int
+        Number of compounds to be put in box.
+    density : float, units kg/m^3, default=None
+        Target density for the sphere in macroscale units.
+    overlap : float, units nm, default=0.2
+        Minimum separation between atoms of different molecules.
+    seed : int, default=12345
+        Random seed to be passed to PACKMOL.
+    edge : float, units nm, default=0.2
+        Buffer at the edge of the sphere to not place molecules. This is necessary
+        in some systems because PACKMOL does not account for periodic boundary
+        conditions in its optimization.
+    compound_ratio : list, default=None
+        Ratio of number of each compound to be put in sphere. Only used in the
+        case of `density` having been specified, `n_compounds` not specified, 
+        and more than one `compound`.
+    fix_orientation : bool or list of bools
+        Specify that compounds should not be rotated when filling the sphere,
+        default=False.
+    temp_file : str, default=None
+        File name to write PACKMOL's raw output to.
+
+    Returns
+    -------
+    filled : mb.Compound
+
+    """
+    _check_packmol(PACKMOL)
+
+    arg_count = 2 - [n_compounds, density].count(None)
+    if arg_count != 1:
+        msg = ("Exactly 1 of `n_compounds` and `density` "
+               "must be specified. {} were given.".format(arg_count))
+        raise ValueError(msg)
+
+    if isinstance(sphere, (list, set)):
+        if len(sphere) != 4:
```

Comment:
tuple should also be fine. might even be able to use `list(sphere)` to assert that it is the type you want